flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/sink/KafkaRecordSerializationSchema.java

```diff
@@ -85,4 +86,28 @@ default void open(
          */
         int[] getPartitionsForTopic(String topic);
     }
+
+    /**
+     * Creates a default schema builder to provide common building blocks i.e. key serialization,
+     * value serialization, partitioning; based on a static target topic.
+     *
+     * @param topic target topic for all records
+     * @param <T> type of incoming elements
+     * @return {@link KafkaRecordSerializationSchemaBuilder}
+     */
+    static <T> KafkaRecordSerializationSchemaBuilder<T> builder(String topic) {
+        return new KafkaRecordSerializationSchemaBuilder<T>(topic);
+    }
+
+    /**
+     * Creates a default schema builder to provide common building blocks i.e. key serialization,
+     * value serialization, partitioning; based on a dynamic target topic.
+     *
+     * @param topicSelector to dynamically derive the target topic based on the incoming element
+     * @param <T> type of incoming elements
+     * @return {@link KafkaRecordSerializationSchemaBuilder}
+     */
+    static <T> KafkaRecordSerializationSchemaBuilder<T> builder(Function<T, String> topicSelector) {
+        return new KafkaRecordSerializationSchemaBuilder<T>(topicSelector);
+    }
```

Comment:
Maybe just start with `builder()`? With only one parameter, it looks like it's the only mandatory parameter.

On a second thought, it could actually be the only parameter, if the builder uses the Flink Type system to serialize to `byte[]` by default. But that requires some small shenanigans with passes the type information through some internal functions in `DefaultKafkaSinkContext`...

Reply:
I am not sure I fully understand that comment. Is this an idea or a proper suggestion? :)

Reply:
The second part was an idea that if you don't pick up on it would still mean that I  would get rid of the two `builder` and collapse to one because of consistency. (value serializer mandatory but not a parameter here)